### ISSUE TITLE
Build fail for project using MLPack, M_SQRT2 undeclared identifier

### ISSUE THREAD

**Issue by rmalde** [mlpack/mlpack#3561]
Hi, I'm trying to use MLPack 4.2.1 in my C++ project. VS 2017. 

I followed the instructions on https://github.com/mlpack/mlpack/blob/master/doc/user/build_windows.md to build MLPack (it built with no errors), and then linked mlpack, cereal, armadillo, and ensmallen in my project properties. 

Now, I'm trying to include in my project:
`#include <mlpack.hpp>`

and I get the build error  `M_SQRT2 undeclared identifier` for `lin_alg_impl.hpp`. 
It looks like this should be solved by importing cmath, which happens in `mlpack/prereqs.hpp`, so I'm not sure why it's not identifying this.

**Comment by rcurtin** [comment]
Yeah, cmath is the first import after Armadillo in mlpack (and Armadillo itself should import cmath).  Is it possible your project defines `MLPACK_HPP` somewhere?  That is the name of the header guard used for `mlpack.hpp`.  Next it would be worth checking what happens if you include `<cmath>` before mlpack.  If that doesn't work, then it seems like something deeper may be wrong.

(Could it be a precompiled header issue?  I'm not sure exactly what MSVC does with those.)

**Comment by rmalde** [comment]
There's no `MLPACK_HPP` or anything related defined anywhere in my project.

Yeah I tried including `<cmath>` in my project before mlpack, still get the same error. The weird thing is, when I do this, I can use `M_SQRT2` just fine in my own project. But for some reason it's not getting imported for the mlpack headers. 

I got my mlpack 4.2.1.tar.gz from the website, and I haven't edited anything, I verified that has the include cmath line. 

Idk if people do this, but happy to get on a zoom call to debug if that's easier. I usually code in Python so it could be an obvious c++ error that I'm oblivious to.

**Comment by rmalde** [comment]
Ok it turns out my error was that I was importing other things before `mlpack.hpp`, when I changed the order so that mlpack is imported first, it worked ¯\_(ツ)_/¯
